scripts/report_scripts/variantreport_p_sample.Rmd

```diff
@@ -525,59 +525,68 @@ Called SNVs:
 Variant effect prediction (*Ensemble VEP output*):  
 [Download VEP_out.csv](`r sprintf('data:text/csv;base64,%s', vep)`)
 
-```{r csv_output_variant_plot, include = F, eval = executeDeconvolution}
+```{r csv_output_variant_plot, include = F}
 # prepare processed variant values to output them as a csv which will be used for the plots in index.rmd
-# those outputs are not offically declared as outputs which can lead to issues - that part should be handled by a seperate 
+# those outputs are not offically declared as outputs which can lead to issues - that part should be handled by a seperate
 # file (and maybe rule)
-# get all possible variants
-all_variants <- colnames (msig_simple[, -which( names(msig_simple) %in% "muts" )])
-output_variants <- file.path(csv_output_dir, "data_variant_plot.csv")
-# 1. write dataframe with this information here
-output_variant_plot <- data.frame (samplename = character(),
+output_variants <- file.path(variants_output_dir, paste0(sample_name, "_variants.csv"))
+output_variant_plot <- data.frame(samplename = character(),
                                    dates = character(),
                                    location_name = character(),
                                    coordinates_lat = character(),
                                    coordinates_long = character())
-# add columns for all possible variants to the dataframe
-for (variant in all_variants) {
-  output_variant_plot[,variant] <- numeric()
-}
-meta_data <- c (samplename = sample_name,
-                dates = date,
-                location_name = location_name,
-                coordinates_lat = coordinates_lat,
-                coordinates_long = coordinates_long)
-
-output_variant_plot <- bind_rows(output_variant_plot, meta_data)
-
-# get rownumber for current sample
-sample_row <- which(grepl(sample_name, output_variant_plot$samplename))
-
-# write mutation frequency values to df
-for (i in all_variants){
-  if (i %in% df$name){
-    # check if variant already has a column
-    if (i %in% colnames(output_variant_plot)) {
-        output_variant_plot[sample_row,][i] <- df$value[df$name == i]
-        output_variant_plot <- output_variant_plot %>% mutate(others = 1-rowSums(across(all_of(all_variants)), na.rm = TRUE))
-}
+if (!executeDeconvolution) {
```

Comment:
oh, answered my question from above.

Please invert the if condition.  When using if / else always aim to use a positive condition.